File: src/main/java/br/com/rotaractsorocabaleste/rotasales/user/UserController.java
package br.com.rotaractsorocabaleste.rotasales.user;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequestMapping("/users")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;
    private final PasswordEncoder passwordEncoder;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public User create(@RequestBody @Valid final User user) {
        return userService.create(user.withPassword(passwordEncoder.encode(user.getPassword())));
    }

    @PatchMapping
    public void changePassword(@RequestBody @Valid final ChangePasswordDTO changePasswordDTO) {
        userService.changePassword(changePasswordDTO.withPassword(passwordEncoder.encode(changePasswordDTO.getPassword())));
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/entrypoint/controller/UserController.kt
package br.com.rotaractsorocabaleste.rotasales.entrypoint.controller

import br.com.rotaractsorocabaleste.rotasales.core.entity.User
import br.com.rotaractsorocabaleste.rotasales.core.service.UserService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController(
    private val userService: UserService,
    private val passwordEncoder: PasswordEncoder,
    private val logger: Logger = LoggerFactory.getLogger(UserController::class.java)
) {

    @PostMapping
    fun createUser(@RequestBody user: User): ResponseEntity<User> {
        logger.info("Received request for create a user, username=${user.username}")

        val ret = userService.save(
            user.copy(
                password = passwordEncoder.encode(user.password)
            )
        )

        logger.info("New user saved, username=${ret.username}")

        return ResponseEntity.ok(ret)
    }

}